main.tf

```diff
@@ -100,9 +100,10 @@ module "lambda" {
   # Do not use Lambda's policy for cloudwatch logs, because we have to add a policy
   # for KMS conditionally. This way attach_policy_json is always true independenty of
   # the value of presense of KMS. Famous "computed values in count" bug...
-  attach_cloudwatch_logs_policy = false
-  attach_policy_json            = true
-  policy_json                   = element(concat(data.aws_iam_policy_document.lambda[*].json, [""]), 0)
+  attach_cloudwatch_logs_policy     = false
+  attach_policy_json                = true
+  policy_json                       = element(concat(data.aws_iam_policy_document.lambda[*].json, [""]), 0)
+  use_existing_cloudwatch_log_group = aws_cloudwatch_log_group.lambda.name != "" ? true : false
```

Comment:
We need to wait for the log group to be created first because the lambda module fetches this using a data source:

```
data "aws_cloudwatch_log_group" "lambda" {
  count = var.create && var.create_function && ! var.create_layer && var.use_existing_cloudwatch_log_group ? 1 : 0

  name = "/aws/lambda/${var.lambda_at_edge ? "us-east-1." : ""}${var.function_name}"
}
```